src/Native/Unix/System.Security.Cryptography.Native/pal_dsa.cpp

```diff
@@ -45,12 +45,34 @@ extern "C" int32_t CryptoNative_DsaSizeSignature(DSA* dsa)
 
 extern "C" int32_t CryptoNative_DsaSizeP(DSA* dsa)
 {
-    return BN_num_bytes(dsa->p);
+    if (dsa)
+    {
+        const BIGNUM* p;
+        DSA_get0_pqg(dsa, &p, NULL, NULL);
```

Comment:
Since the 2.1 and 2.2 branches are C++-based, these `NULL`s should probably all be `nullptr`s (applies to all lines added in this change).

Reply:
Should be fixed now.